Source/CompatibilityPatches/DoorsExpanded.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Harmony;
+using RimWorld;
+using Verse;
+
+namespace Locks.CompatibilityPatches
+{
+    public class DoorsExpandedPatches
+    {
+        private static bool CanLock(BuildableDef def)
+        {
+            return def.stuffCategories
+                       .Count(category =>
+                           category.defName == StuffCategoryDefOf.Fabric.defName 
+                           || category.defName == StuffCategoryDefOf.Leathery.defName) == 0;
+        }
+
+        private static void Init()
+        {
+            var doorExpandedType = Type.GetType("DoorsExpanded.Building_DoorExpanded, DoorsExpanded");
+            var harmony = HarmonyInstance.Create("Harmony_Locks_DoorsExpanded");
+            
+            harmony.Patch(
+                doorExpandedType.GetMethod("PawnCanOpen", BindingFlags.Public | BindingFlags.Instance),
+                new HarmonyMethod(typeof(Patch_Building_DoorExpanded_PawnCanOpen).GetMethod(nameof(Patch_Building_DoorExpanded_PawnCanOpen.Prefix)))); 
+            harmony.Patch(
+                doorExpandedType.GetMethod("GetGizmos", BindingFlags.Public | BindingFlags.Instance),
+                null, 
+                new HarmonyMethod(typeof(Patch_Building_DoorExpanded_GetGizmos).GetMethod(nameof(Patch_Building_DoorExpanded_GetGizmos.Postfix))));
+            
+            DefDatabase<ThingDef>.AllDefsListForReading.ForEach(def =>
+            {
+                if (def != null && def.thingClass == doorExpandedType)
+                {
+                    if (CanLock(def))
+                    {
+                        Initialization.AddLock(def);
+                    }
+                }
+            });
+        }
+        
+        [HarmonyPatch(typeof(DefGenerator), "GenerateImpliedDefs_PostResolve")]
+        public class Patch_GenerateImpliedDefs_PostResolve
+        {
+            [HarmonyPriority(Priority.Last)]
+            public static void Postfix()
+            {
+                if (ModsConfig.ActiveModsInLoadOrder.Any(m => m.Name == "Doors Expanded"))
+                {
+                    Log.Message("Lock Mods detected Doors Expanded, loading support");
+                    try
+                    {
+                        ((Action) (() => { Init(); }))();
```

Comment:
This is for depending on mod assemblies that might not be present at runtime. How to use:

- Wrap the call in an action and call it immediately while checking for `TypeLoadException`
- Don't have any `using` declarations referring to possibly missing types, use fully qualified names instead